ee/control/consumers/flareconsumer/flareconsumer.go

```diff
@@ -0,0 +1,56 @@
+package flareconsumer
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/kolide/launcher/pkg/debug/checkups"
+	"github.com/kolide/launcher/pkg/debug/shipper"
+)
+
+const (
+	// Identifier for this consumer.
+	FlareSubsystem = "flare"
+)
+
+type FlareConsumer struct {
+	flarer   flarer
+	knapsack types.Knapsack
+	// newFlareStream is assigned to a field so it can be mocked in tests
```

Comment:
maybe do the same with flarer so the caller doesn't have to instantiate the flare runner